app/src/main/java/com/example/sharingang/DetailedItemFragment.kt

```diff
@@ -67,20 +71,22 @@ class DetailedItemFragment : Fragment() {
         binding.shareButton.setOnClickListener { shareItem() }
 
         viewModel.setUser(args.item.userId)
-        viewModel.user.observe(viewLifecycleOwner, { user ->
-            binding.username = "Posted by ${user?.name}"
-            binding.itemPostedBy.visibility = if (user != null) View.VISIBLE else View.GONE
-            binding.itemPostedBy.setOnClickListener { view ->
-                view.findNavController().navigate(
-                    DetailedItemFragmentDirections.actionDetailedItemFragmentToUserProfileFragment(
-                        user?.id
-                    )
-                )
-            }
-        })
+        viewModel.user.observe(viewLifecycleOwner, this::onUserChange)
         return binding.root
     }
 
+    private fun onUserChange(user: User?) {
+        binding.username = "Posted by ${user?.name}"
```

Comment:
Is there any way you can use a resource string here? In case we decide to change the language for example, we wouldn't need to go through the code.

Reply:
I didn't add this code, I just formatted the file